feat: add bash/zsh completion for p11* tools

mdevolde opened the pull request:
This PR adds bash completion support for several `p11*` CLI tools.

### Features
- Shared logic in `completion/p11-common`
- Specific completions for: `p11cat`, `p11cp`, `p11kcv`, `p11ls`, `p11more`, `p11mv`, `p11od`, `p11setattr`
- Supports `PKCS11LIB`, `PKCS11SLOT`, `PKCS11PASSWORD` as environment fallback
- Documented usage, environment handling, and limitations

### Makefile
- Adds completion scripts to `EXTRA_DIST`
- Installs them to `${datadir}/bash-completion/completions`

### Limitations
- Does not support inline env variables (e.g. `PKCS11LIB=... p11cat`)
- Zsh is not yet supported, could be addressed in a later contribution

keldonin commented on completion/p11-common:
It would be great to have support for slot selection. That would require to modify `p11slotinfo` and add an option to enumerate slots and token, for completion.

keldonin reviewed (changes requested):
Good stuff. 

A few requests:
- It should support ZSH as well
- If possible, infer slots as well - that would require to adapt `p11slotinfo`to yield a list of slots and token, for completion.

mdevolde committed:
feat: add bash/zsh completion for p11* tools

mdevolde committed:
feat: add bash/zsh completion for p11* tools

mdevolde commented:
I added:
- Completion on all `p11` commands
- Slot completion
- Token label completion
- Added a dedicated func in `p11slotinfo` to handle slot and token completion
- Completion of `-i` and `-w` parameters with PKCS11 objects
- Consideration of token labels for PKCS11 object completion
- Added support for `PKCS11TOKENLABEL` as env fallback 
- Basic file completion for relevant parameters (e.g., `-l` or `-o`)
- Completion of certain parameters that only accept a predefined list of strings (such as `-k` on `p11keygen`, for example).
- Completion of parameters after writing a `-`.

I did not add completion for:
- `-b` (`p11kcv`, `p11keygen`), `-n` (`p11kcv`), `-c` (`p11keycomp`), `-u` (`p11mkcert`), `-e` (`p11keygen`) -> These parameters are just numbers, for which completion is not relevant.
- `-p` (all), `-J` (`p11keygen`, `p11rewrap`, `p11wrap`) -> These parameters are PINs or tokens, for which completion is not possible.
- `-d` (`p11mkcert`, `p11req`), `-W` (`p11keygen`, `p11rewrap`, `p11wrap`) -> Since the user must enter a custom and structured string himself, completion is not relevant.

I wonder if it is relevant to do completion for:
- `-m` (all) -> Should I just suggest completion with directories, or should I suggest completion with NSS prefixes before directories?
-  `-a` (`p11wrap`) -> There is a given list of accepted algorithms, but some accept parameters, and these parameters cannot be completed. Should I just provide the algorithms, without the parameters?
- `-e` (`p11mkcert`, `p11req`) -> There are three accepted prefixes (`DNS`, `email`, `IP`), followed by a value decided by the user. Should I provide completion for these prefixes?

For now, it's still just completion on bash, not zsh.

I also fixed an error in the `README`, where an obsolete environment variable was listed.

mdevolde commented:
Since my last update, I have:
- Added directory completion for `-m`.
- Added completion for `-a` algorithms (`p11wrap`), without adding extra space after, to allow for opening parentheses and specifying parameters.
- Added prefix completion for `-e` (`p11mkcert`, `p11req`), without adding an extra space after, to allow specifying a value.

I also added ZSH compatibility for completions.
Currently, bash completions are used by ZSH. To get more advanced ZSH completion features, such as descriptions, you need to write additional dedicated scripts.

mdevolde commented:
Noticed and fixed a compatibility bug for ZSH with `bashcompinit` (`local -n`).

covertmatthew commented on lib/pkcs11_slotinfo.c:
```diff
@@ -245,5 +245,99 @@ func_rc pkcs11_info_ecsupport(pkcs11Context *p11Context)
+    CK_SLOT_INFO slotInfo;
+    CK_TOKEN_INFO tokenInfo;
+    CK_ULONG ulSlotCount = 0;
+    int i;
```
Fix the `-Wsign-compare` warning in `lib/pkcs11_slotinfo.c` (`pkcs11_enumerate_slots_with_token()` declares `int i;` but compares it against `CK_ULONG ulSlotCount`. Declaring `i` as `CK_ULONG` should clear it.

mdevolde replied:
done

covertmatthew commented on src/p11slotinfo.c:
```diff
@@ -87,6 +88,8 @@ int main( int argc, char ** argv )
     char * tokenlabel = NULL;
     int ec_support = 0;
     int so=0;
+	int list_slots=0;
```
The new lines (`int list_slots=0;` and the `if (list_slots) { … } else { … }` block) mix tabs with the file's existing space indentation. Please re-indent with spaces to match the surrounding code.

mdevolde replied:
done

covertmatthew reviewed (commented):
@mdevolde updates look good. Verbally approving, but I'd like @keldonin to formally approve before merging.
